index.html

```
@@ -1238,6 +1238,192 @@
+
+  </ol>
+
+  <p class="note">This algorithm returns an object with a single property that is an array, instead
+  of just returning an array, so that it is possible to extend the algorithm in the future if
+  necessary.</p>
+
+  <p>When the user agent is to <dfn>get the object</dfn> for an item <var>item</var>,
+  potentially together with a list of elements <var>memory</var>, it must run the following
+  substeps:</p>
+
+  <ol>
+
+   <li><p>Let <var>result</var> be an empty object.</p></li>
+
+   <li><p>If no <var>memory</var> was passed to the algorithm, let <var>memory</var> be an empty list.</p></li>
+
+   <li><p>Add <var>item</var> to <var>memory</var>.</p></li>
+
+   <li><p>If the <var>item</var> has any <a>item types</a>, add an entry to <var>result</var>
+   called "<code>@type</code>" whose value is an array listing the
+   <a>item types</a> of <var>item</var>, in the order they were specified on the
+   <code><a>itemtype</a></code> attribute.</p>
+
+   </li><li><p>If the <var>item</var> has a <a>global identifier</a>, add an entry to
+   <var>result</var> called "<code>@id</code>" whose value is the <a>global
+   identifier</a> of <var>item</var>.</p></li>
+
+   <li><p>Let <var>properties</var> be an empty object.</p></li>
+
+   <li>
+
+    <p>For each element <var>element</var> that has one or more <a>property names</a>
+    and is one of <a data-lt="item properties">the properties of the item</a>
+    <var>item</var>, in the order those elements are given by the algorithm that returns
+    <a data-lt="item properties">the properties of an item</a>, run the following substeps:</p>
+
+    <ol>
+
+     <li><p>Let <var>value</var> be the
+     <a>property value</a> of <var>element</var>.</p></li>
+
+     <li><p>If <var>value</var> is an <a data-lt="concept item">item</a>, then:
+     If <var>value</var> is in <var>memory</var>, then let <var>value</var> be
+     the string "<code>ERROR</code>". Otherwise, <a>get the object</a> for
```

It feels strange to set a value to the string `ERROR`... feels like you should throw an exception or do something less extreme than throwing but more extreme than returning an 'ERROR' string.

---

Actually, in this case, shouldn't it be a node reference? After all, the value is an item, and we should just reference that item.